### PR TITLE
:hammer: Add devcontainer

### PR DESCRIPTION
**Pull request by cyw3**
Provide a develop enviroment docker container.

Developers can use it and don't need to deploy the build environment again.

### CODE REVIEW

**Comment by mpollmeier** [comment]
What @ursachec said. Thank you for sharing, but we have a few questions. 
It would be good to get a better description of the use case, ideally in the readme, e.g. in a development section. 

The base image is almost 1g, comes from a vendor and ships their jdk:
```
[root@cf01afb1383b TencentKona-11.0.15.b1]# bin/java -version
openjdk version "11.0.15-ga" 2022-04-28
OpenJDK Runtime Environment TencentKonaJDK (build 11.0.15-ga+1)
OpenJDK 64-Bit Server VM TencentKonaJDK (build 11.0.15-ga+1, mixed mode)
```

I don't know much about them, but can't we just use a more common/standard and/or small image like openjdk or some alpine linux?

**Comment by cyw3** [comment]
@ursachec 

It is easy to deploy the build environment with devcontainer and VSCode extension `remote - container`. And developers don't need to care about the build environment again and can immediately focus on their work.

Compared to such benefits, the cost of `the choice of the base image and the JDK distribution` is relatively small.

**Comment by cyw3** [comment]
> What @ursachec said. Thank you for sharing, but we have a few questions. It would be good to get a better description of the use case, ideally in the readme, e.g. in a development section.
> 
> The base image is almost 1g, comes from a vendor and ships their jdk:
> 
> ```
> [root@cf01afb1383b TencentKona-11.0.15.b1]# bin/java -version
> openjdk version "11.0.15-ga" 2022-04-28
> OpenJDK Runtime Environment TencentKonaJDK (build 11.0.15-ga+1)
> OpenJDK 64-Bit Server VM TencentKonaJDK (build 11.0.15-ga+1, mixed mode)
> ```
> 
> I don't know much about them, but can't we just use a more common/standard and/or small image like openjdk or some alpine linux?

`TencentOS` and `TencentKona` are just the software I use more often. 

If you have a better recommendation, I can try to modify it here.

**Comment by mpollmeier** [comment]
yes, centos and/or openjdk sound good

**Comment by cyw3** [comment]
Ok, I update the devcontainer with them now.

**Comment by mpollmeier** [comment]
Thank you! Now would you mind adding a few words on how to use it, e.g. in the readme? That'll also give it visibility to others.

**Comment by cyw3** [comment]
> Thank you! Now would you mind adding a few words on how to use it, e.g. in the readme? That'll also give it visibility to others.

DONE

**Comment by mpollmeier** [comment]
Thank you! I'll merge this as is and will add a section on Intellij, which is the IDE most of our team use. Good to get an overview and instructions for different dev setups.

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+	"name": "dev environment",
+	"dockerFile": "Dockerfile",
+	"extensions": [
+		"vscjava.vscode-java-pack",
+		"scala-lang.scala",
+		"scalameta.metals",
+		"scala-lang.scala-snippets",
+		"eamodio.gitlens",
```
Comment (ursachec) [inline .devcontainer/devcontainer.json:9]: This extension does not seem relevant to the change, please remove it
```suggestion
```

Comment (cyw3) [reply .devcontainer/devcontainer.json]: done

**Comment by cyw3** [comment]
done

**Review by mpollmeier — APPROVED** [review]
